internal/testing/validation/postgrescluster_test.go

```diff
@@ -6,515 +6,16 @@ package validation
-		instances: [{
-			dataVolumeClaimSpec: {
-				accessModes: [ReadWriteOnce],
-				resources: { requests: { storage: 1Mi } },
-			},
-		}],
-	}`)
-
-	base.Namespace = namespace.Name
-	base.Name = "postgres-user-options"
-
-	assert.NilError(t, cc.Create(ctx, base.DeepCopy(), client.DryRunAll),
-		"expected this base cluster to be valid")
-
-	// See [internal/controller/postgrescluster.TestValidatePostgresUsers]
-
-	t.Run("NoComments", func(t *testing.T) {
-		cluster := base.DeepCopy()
-		cluster.Spec.Users = []v1beta1.PostgresUserSpec{
-			{Name: "dashes", Options: "ANY -- comment"},
-			{Name: "block-open", Options: "/* asdf"},
-			{Name: "block-close", Options: " qw */ rt"},
-		}
-
-		err := cc.Create(ctx, cluster, client.DryRunAll)
-		assert.Assert(t, apierrors.IsInvalid(err))
-		assert.ErrorContains(t, err, "cannot contain comments")
-
-		status := require.StatusError(t, err)
-		assert.Assert(t, status.Details != nil)
-		assert.Assert(t, cmp.Len(status.Details.Causes, 3))
-
-		for i, cause := range status.Details.Causes {
-			assert.Equal(t, cause.Field, fmt.Sprintf("spec.users[%d].options", i))
-			assert.Assert(t, cmp.Contains(cause.Message, "cannot contain comments"))
-		}
-	})
-
-	t.Run("NoPassword", func(t *testing.T) {
-		cluster := base.DeepCopy()
-		cluster.Spec.Users = []v1beta1.PostgresUserSpec{
-			{Name: "uppercase", Options: "SUPERUSER PASSWORD ''"},
-			{Name: "lowercase", Options: "password 'asdf'"},
-		}
-
-		err := cc.Create(ctx, cluster, client.DryRunAll)
-		assert.Assert(t, apierrors.IsInvalid(err))
-		assert.ErrorContains(t, err, "cannot assign password")
-
-		status := require.StatusError(t, err)
-		assert.Assert(t, status.Details != nil)
-		assert.Assert(t, cmp.Len(status.Details.Causes, 2))
-
-		for i, cause := range status.Details.Causes {
-			assert.Equal(t, cause.Field, fmt.Sprintf("spec.users[%d].options", i))
-			assert.Assert(t, cmp.Contains(cause.Message, "cannot assign password"))
-		}
-	})
-
-	t.Run("NoTerminators", func(t *testing.T) {
-		cluster := base.DeepCopy()
-		cluster.Spec.Users = []v1beta1.PostgresUserSpec{
-			{Name: "semicolon", Options: "some ;where"},
-		}
-
-		err := cc.Create(ctx, cluster, client.DryRunAll)
-		assert.Assert(t, apierrors.IsInvalid(err))
-		assert.ErrorContains(t, err, "should match")
-
-		status := require.StatusError(t, err)
-		assert.Assert(t, status.Details != nil)
-		assert.Assert(t, cmp.Len(status.Details.Causes, 1))
-		assert.Equal(t, status.Details.Causes[0].Field, "spec.users[0].options")
-	})
-
-	t.Run("Valid", func(t *testing.T) {
-		cluster := base.DeepCopy()
-		cluster.Spec.Users = []v1beta1.PostgresUserSpec{
-			{Name: "normal", Options: "CREATEDB valid until '2006-01-02'"},
-			{Name: "very-full", Options: "NOSUPERUSER NOCREATEDB NOCREATEROLE NOINHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT 5"},
-		}
-
-		assert.NilError(t, cc.Create(ctx, cluster, client.DryRunAll))
-	})
-}
-
 func TestPostgresUserInterfaceAcrossVersions(t *testing.T) {
```

Comment:
Would it be worth renaming this file in the future? I like the specificity of the new files' names.